arc/cli/commands/recommend.py

```
@@ -199,253 +222,333 @@ async def _save_recommendations_to_database(db_client, run_result, recommendatio
+        "llm_analysis": {
+            "confidence_score": response.confidence_score,
+            "processing_time": response.processing_time,
+            "analysis_summary": response.analysis_summary,
+            "total_recommendations": len(response.recommendations)
+        }
+    }
+    
+    return llm_recommendations
+
+
+def _build_model_optimization(run_result, model_performance_data: Optional[dict] = None):
```

Ruff (0.11.9)

Use `X | Y` for type annotations

Convert to `X | Y`

(UP007)